3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -20,18 +20,62 @@ Use async/await and try/catch to handle promises.
 Try and avoid using global variables. As much as possible, try and use function 
 parameters and return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+
+async function fetchData(url) {
+  const fetchData = await fetch(url);
+  const data = await fetchData.json();
+  return data.results;
 }
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function fetchAndPopulatePokemons(array) {
+  let index = 0;
+  array.forEach((pokemon) => {
+    const option = document.createElement('option');
+    option.textContent = pokemon.name;
+    option.value = index;
```

Comment:
Instead of assigning an index value to the `option.value` it is better to use `pokemon.url`. In that way you can access that URL inside the `change` event handler directly, without having to fetch the data again. (And you can get rid of the `index` variable).

```suggestion
    option.value = pokemon.url;
```